src/Datadog.Trace/Agent/Api.cs

```diff
@@ -38,27 +37,20 @@ static Api()
             };
         }
 
-        public Api(Uri baseEndpoint, DelegatingHandler delegatingHandler, IStatsd statsd)
+        public Api(Uri baseEndpoint, DelegatingHandler delegatingHandler, IStatsd statsd, int? overridingPort = null)
```

Comment:
The caller knows the value of `overridingPort`, so it can replace the port in `baseEndpoint` before passing it in. This is not `Api`'s concern.